lnrpc/routerrpc/router_backend.go

```diff
@@ -258,7 +256,85 @@ func (r *RouterBackend) parseQueryRoutesRequest(in *lnrpc.QueryRoutesRequest) (
 	if sourcePubKey != r.SelfNode {
 		maxTotalTimelock = math.MaxUint32
 	}
-	cltvLimit, err := ValidateCLTVLimit(in.CltvLimit, maxTotalTimelock)
+
+	// If we have a blinded path set, we'll get a few of our fields from
+	// inside of the path rather than the request's fields.
+	var (
+		cltvLimit      = in.CltvLimit
+		targetPubKey   *route.Vertex
+		routeHintEdges map[route.Vertex][]*channeldb.CachedEdgePolicy
+		blindedPmt     *routing.BlindedPayment
+	)
+
+	// Validate that the fields provided in the request are sane depending
+	// on whether it is using a blinded path or not.
+	if len(in.BlindedPath) > 0 { //nolint:nestif
+		if len(in.PubKey) != 0 {
+			return nil, fmt.Errorf("target pubkey: %x should "+
+				"not be set when blinded path is provided",
+				in.PubKey)
+		}
+
+		if len(in.BlindedPath) != 1 {
+			return nil, errors.New("query routes only supports " +
+				"a single blinded path")
+		}
+
+		var (
+			blindedPath  = in.BlindedPath[0]
+			limitSet     = in.CltvLimit != 0
+			blindedLimit uint32
+		)
+
+		if blindedPath.RelayConstraints != nil {
+			blindedLimit = blindedPath.RelayConstraints.CltvLimit
+		}
+
+		if limitSet && in.CltvLimit != blindedLimit {
+			return nil, fmt.Errorf("cltv limit in request: %v "+
+				"different to blinded path: %v", in.CltvLimit,
+				blindedLimit)
+		}
+		cltvLimit = blindedLimit
```

Comment:
so my understanding says that this blindedPath.RelayConstraints.CltvLimit should refer only to the limit inside the blinded path? Though why should we limit our CLTV-Limit for the other part of the route (the non-blinded part)?

Reply:
I think this was based on an old version of the spec - no cltv limit for blinded path payments in the merged version, now they're _just_ in the payloads for the blinded hops so that they can "expire" the blinded route to prevent probing.

Reply:
ohh thanks for the explanation now it makes sense

Reply:
Ok I have 2 question then:

1. Where do we get a reasonable `TotalCltvDelta` for the BlindedPayment, in prior Spec Version it seemed to me it was part of the Aggregated Data but now it seems we decide it by ourself? Feels not quite right or ?

Reply:
1. Total cltv delta for the blinded hops and blinded destination = `QueryRoutesRequest.BlindedPath.TotalCltvDetla`* + `QueryRoutes.Request.FinalCltvDelta` (The `FinalCltvDelta` would be obtained from the invoice containing the blinded path. 
2. This is already handled in pathfinding - the [finalHTLCExpiry](https://github.com/carlaKC/lnd/blob/bdb41e58677836390f817b481fd784148f310571/routing/router.go#L1778) is [provided to pathfinding](https://github.com/carlaKC/lnd/blob/bdb41e58677836390f817b481fd784148f310571/routing/router.go#L1793) and [added](https://github.com/lightningnetwork/lnd/blob/e549096b8837a95e0515c8c63fa01c634714d913/routing/pathfind.go#L613) to our `cltvLimit`.

*Looking at this again - I think that `TotalCltvDelta` in `BlindedPayment` is a bad name - `BlindedCltvDelta`?